packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2138,6 +2125,118 @@ export class MergeTree {
         return segRef;
     }
 
+    private ensureLocalPartialsComputed(refSeq: number): void {
+        if (!this.localPartialsComputed) {
+            const rebaseCollabWindow = new CollaborationWindow();
+            rebaseCollabWindow.loadFrom(this.collabWindow);
+            if (refSeq < this.collabWindow.minSeq) {
+                rebaseCollabWindow.minSeq = refSeq;
+            }
+            this.root.partialLengths = PartialSequenceLengths.combine(
+                this.root,
+                rebaseCollabWindow,
+                true,
+                true,
+            );
+            this.localPartialsComputed = true;
+        }
+    }
+
+    /**
+     * Normalizes the segments nearby `segmentGroup` to be ordered as they would if the op submitting `segmentGroup`
+     * is rebased to the current sequence number.
+     * This primarily affects the ordering of adjacent segments that were removed between the original submission of
+     * the local ops and now.
+     * Consider the following sequence of events:
+     * Initial state: "hi my friend" (seq: 0)
+     * - Client 1 inserts "good " to make "hi my good friend" (op1, refSeq: 0)
+     * - Client 2 deletes "my " to make "hi friend" (op2, refSeq: 0)
+     * - op2 is sequenced giving seq 1
+     * - Client 1 disconnects and reconnects at seq: 1.
+     *
+     * At this point in time, client 1 will have segments ["hi ", Removed"my ", Local"good ", "friend"].
+     * However, the rebased op that it submits will cause client 2 to have segments
+     * ["hi ", Local"good ", Removed"my ", "friend"].
+     *
+     * The difference in ordering can be problematic for tie-breaking concurrently inserted segments in some scenarios.
+     * Rather than incur extra work tie-breaking these scenarios for all clients, when client 1 rebases its operation,
+     * it can fix up its local state to align with what would be expected of the op it resubmits.
+     */
+    public normalizeSegmentsOnRebase(segmentGroup: SegmentGroup): void {
+        // Forward and backward excursions will expand the affected normalization range,
+        // so simply taking the first segment suffices
+        const { refSeq, segments: [segment] } = segmentGroup;
+        const forwardLocalSegs: ISegment[] = [];
```

Comment:
could we get by with a single list here, insert the original seg, and unshift backward, and push forward, then we don't have to enumerate and build another array to get to consecutiveLocalSegments